test/e2e/specs/all-features-confirmation/record/presentation.spec.js

```diff
@@ -163,4 +172,72 @@ describe('View existing record,', function() {
+              // related table should be visible
+                expect(rtTableHeading.isDisplayed()).toBeTruthy("Category_5 heading is not visible.");
+                return rtTableHeading.getAttribute("class");
+            }).then (function(className) {
+                expect(className).toContain("panel-open", "Related table panel is not open when clicked through TOC.");
+            }).catch( function(err) {
+                console.log(err);
+                expect('Encountered an error').toBe('Please check the log', 'Inside catch block');
+            });
+        });
+
+        it('Record count along with heading should match for the panel and related table content should be in correct order', function(){
+            chaisePage.recordPage.getSidePanelTableTitles().then(function(tableNames){
+              for (var i=0; i<tableNames.length; i++){
+                tableNames[i] = tableNames[i].replace(/ +/g, " ");
+              }
+                expect(tableNames).toEqual(testParams.sidePanelTest.sidePanelTableOrder,"Order is not maintained for related tables in side panel");
+                expect(tableNames.length).toEqual(testParams.sidePanelTest.tocCount, "Count mismatch for number of related tables in side panel");
+                return chaisePage.recordPage.getSidePanelHeading();
+            }).then(function(sidePanelHeading){
+                expect(sidePanelHeading).toBe(testParams.sidePanelTest.panelHeading, "Side Panel heading did not match.");
+            }).catch( function(err) {
+                console.log(err);
+                expect('Encountered an error').toBe('Please check the log', 'Inside catch block');
+            });
+        });
+
+        it('Side panel should hide/show by clicking pull button', function(){
+            var recPan =  chaisePage.recordPage.getSidePanel();
+
+            fiddlerBtn.getAttribute("class").then(function(classNameRight) {
+                expect(classNameRight).toContain('glyphicon glyphicon-triangle-right', 'Side Pan Pull button is not pointing in the right direction');
+                return recPan.getAttribute("class");
+            }).then(function(openPanel){
+                expect(openPanel).toContain('open-panel', 'Side Panel is not visible when fiddler is poining in right direction');
+                fiddlerBtn.click();
+                return fiddlerBtn.getAttribute("class");
+            }).then(function(classNameLeft){
+                expect(classNameLeft).toContain("glyphicon glyphicon-triangle-left", "Side Pan Pull button is not pointing in the left direction.");
+                return recPan.getAttribute("class");
+            }).then(function(closePanel){
+                expect(closePanel).toContain('close-panel', 'Side Panel is not hidden when fiddler is poining in left direction');
+            }).catch( function(err) {
+                console.log(err);
+                expect('Encountered an error').toBe('Please check the log', 'Inside catch block');
```

Comment:
I know that this is what we are doing in most of our test cases . But this is wrong and we should not continue writing it the same way. What you can do is add `done` function as a parameter for the it, and then calling `done.fail()` will fail the test case and `done()` means successful and the end of test case.

```javascript
it ("DESC", function (done) {
   promise1.then(function () {
      return promise2;
   }).then(function () {
      return promise3;
   }).then(function () {
       // ...
      done();
   }).catch (function (err) {
       console.log(err);
       done.fail();
   })
})
```